AndroidCompat/getAndroid.sh

```
@@ -20,15 +20,15 @@ fi
 
 
 # foolproof against running from AndroidCompat dir instead of running from project root
-if [ "$(basename $(pwd))" = "AndroidCompat" ]; then
+if [ "$(basename "$(pwd)")" = "AndroidCompat" ]; then
   cd ..
 fi
 
 
 echo "Getting required Android.jar..."
 rm -rf "tmp"
 mkdir -p "tmp"
-pushd "tmp"
+pushd "tmp" || exit
```

what does this line do?

---

If pushd fails(folder doesnt exist or similar) it will exit the script instead of continuing

---

why would it fail?!

---

Maybe no permissions? It was a lint in IntelliJ